libcore/src/pedestrian/AgentsSource.cpp

```diff
@@ -220,27 +220,20 @@ const std::shared_ptr<StartDistribution> AgentsSource::GetStartDistribution() co
 void AgentsSource::GenerateAgents(std::vector<Pedestrian *> & peds, int count, Building * building)
 {
     std::vector<Point> emptyPositions;
+    // TODO: this vector is empty with purpose!
+    //       1. Source-Pedestrians are created without setting their positions.
+    //       2. In a second step (in AgentsSourcesManager::ProcessAllSources())
+    //          the positions will be calculated and initializes.
+    // TODO: We should reverse the order of points 1 and 2.
     int pid;
     pid = (this->GetAgentId() >= 0) ?
               this->GetAgentId() :
               Pedestrian::GetAgentsCreated() + building->GetAllPedestrians().size();
     for(int i = 0; i < count; i++) {
         if(GetStartDistribution()) {
             auto ped = GetStartDistribution()->GenerateAgent(building, &pid, emptyPositions);
-            if(ped->FindRoute() == -1) {
```

Comment:
`FindRoute` is now only called after the position is set?

Reply:
Yes, in `AdjustVelocityUsingWeidmann(ped)` called in `AgentsSourcesManager.cpp`